### PR TITLE
Added the changes in mentioned branch and some tests for vad and silero vad

### LINKED ISSUE #241: try/implement new voice activity detection model (silero-vad)
https://github.com/huggingface/candle/blob/main/candle-examples/examples/silero-vad/main.rs

definition of done:
- [ ] can use either current vad or new one (use param in cli like ocr engine)
- [ ] some unit test that show some metrics performance diff

### PR DESCRIPTION
**Pull request by itsdheerajdp**
/claim #241 
Fixes #241 

I added the changes of the branch #250 and some tests for vad and silero vad

### CODE REVIEW

File: screenpipe-audio/src/stt.rs [hunk 1]
```diff
@@ -737,70 +738,35 @@ pub async fn create_whisper_channel(
...
                 Some(input) = input_receiver.recv() => {
                     let timestamp = SystemTime::now()
                         .duration_since(UNIX_EPOCH)
                         .expect("Time went backwards")
                         .as_secs();
 
-                    let transcription_result = if cfg!(target_os = "macos") {
-                        #[cfg(target_os = "macos")]
-                        {
-                            autoreleasepool(|| {
```
Comment (louis030195) [inline screenpipe-audio/src/stt.rs:761]: @itsdheerajdp autoreleasepool is necessary https://github.com/huggingface/candle/issues/2271

File: screenpipe-audio/src/stt.rs [hunk 2]
```diff
@@ -810,8 +776,7 @@ pub async fn create_whisper_channel(
                 else => break,
             }
         }
-        // Cleanup code here (if needed)
     });
 
-    Ok((input_sender, output_receiver, shutdown_flag))
```
Comment (louis030195) [inline screenpipe-audio/src/stt.rs:816]: @itsdheerajdp shutdown_flag?

**Comment by louis030195** [comment]
@itsdheerajdp it seems you did not rebase on main, can you do so? then will test, merge, & approve bounty!

**Comment by itsdheerajdp** [comment]
> , can



> @itsdheerajdp it seems you did not rebase on main, can you do so? then will test, merge, & approve bounty!

yeah sure i'll do

**Comment by itsdheerajdp** [comment]
@louis030195 I resolved the changes and also pushed the changes after synching the main branch kindly check it out and let me know if it needs any further updates

File: screenpipe-server/src/core.rs [hunk 3]
```diff
@@ -97,25 +97,16 @@ pub async fn start_continuous_recording(
...
 
-    // Handle any errors from the tasks
     if let Err(e) = video_result {
         error!("Video recording error: {:?}", e);
     }
     if let Err(e) = audio_result {
         error!("Audio recording error: {:?}", e);
     }
 
-    // Shutdown the whisper channel
```
Comment (louis030195) [inline screenpipe-server/src/core.rs:111]: @itsdheerajdp what is this?

File: screenpipe-server/src/core.rs [hunk 4]
```diff
@@ -32,28 +33,27 @@ pub async fn start_continuous_recording(
...
+            CliVadEngine::WebRtc => create_whisper_channel(audio_transcription_engine.clone(), VADDetectionEngine::WebRtc).await?,
+            CliVadEngine::Silero => create_whisper_channel(audio_transcription_engine.clone(), VADDetectionEngine::Silero).await?,
+        }
     };
-    let whisper_sender_clone = whisper_sender.clone();
     let db_manager_video = Arc::clone(&db);
     let db_manager_audio = Arc::clone(&db);
 
-    let is_running_video = Arc::clone(&vision_control);
+    let is_running_video = Arc::clone(&vi   on_control);
```
Comment (louis030195) [inline screenpipe-server/src/core.rs:56]: does this compile?

File: screenpipe-server/src/core.rs [hunk 5]
```diff
@@ -32,28 +33,27 @@ pub async fn start_continuous_recording(
     friend_wearable_uid: Option<String>,
     monitor_id: u32,
     use_pii_removal: bool,
+    vad_engine: CliVadEngine,
 ) -> Result<()> {
-    let (whisper_sender, whisper_receiver, whisper_shutdown_flag) = if audio_disabled {
```
Comment (louis030195) [inline screenpipe-server/src/core.rs:36]: ??

File: screenpipe-server/src/core.rs (hunk 4 above)
Comment (itsdheerajdp) [reply screenpipe-server/src/core.rs:56]: this is typing mistake which can be done while pushing i guess , i resolved this.

File: screenpipe-server/src/core.rs (hunk 5 above)
Comment (itsdheerajdp) [reply screenpipe-server/src/core.rs:36]: I considered your PR(#250) While making changes. But i'll add shutdown flag here also and push the code

**Comment by louis030195** [comment]
@itsdheerajdp please do not request review if you did not resolve my points

**Pull request closed without merging**